src/org/zaproxy/zap/extension/httpsinfo/SSLServer.java

```diff
@@ -114,6 +116,11 @@ public void updateSupportedVersions() {
 
 		if (sh2 != null) {
 			sv.add(0x0200);
+		} else {
+			exceptionCount++;// Count connect failures
+		}
+		if (exceptionCount == 5) {// Failed to connect all 5 times, throw IOException
+			throw new IOException();
```

Comment:
Could use a suitable message instead ("Failed to connect to server after 5 attempts."?)